medium-tests/src/test/java/mediumtest/analysis/AnalysisMediumTests.java

```diff
@@ -650,23 +660,52 @@ void it_should_skip_analysis_only_for_disabled_language(@TempDir Path baseDir) {
   }
 
   @Test
-  void should_trigger_analysis_on_path_to_compile_command_change(@TempDir Path baseDir) {
-    var cppFilePath = createFile(baseDir, "file.cpp", "");
-    var cppFileUri = cppFilePath.toUri();
+  void should_trigger_analysis_on_path_to_compile_command_change(@TempDir Path baseDir) throws IOException {
+    assumeTrue(COMMERCIAL_ENABLED);
+    var cFile = prepareInputFile("foo.c", "#import \"foo.h\"\n", false, StandardCharsets.UTF_8, SonarLanguage.C, baseDir);
+    var cFilePath = baseDir.resolve("foo.c");
+    var buildWrapperContent = "{\"version\":0,\"captures\":[" +
+      "{" +
+      "\"compiler\": \"clang\"," +
+      "\"executable\": \"compiler\"," +
+      "\"stdout\": \"#define __STDC_VERSION__ 201112L\n\"," +
+      "\"stderr\": \"\"" +
+      "}," +
+      "{" +
+      "\"compiler\": \"clang\"," +
+      "\"executable\": \"compiler\"," +
+      "\"stdout\": \"#define __cplusplus 201703L\n\"," +
+      "\"stderr\": \"\"" +
+      "}," +
+      "{\"compiler\":\"clang\",\"cwd\":\"" +
+      baseDir.toString().replace("\\", "\\\\") +
+      "\",\"executable\":\"compiler\",\"cmd\":[\"cc\",\"foo.c\"]}]}";
+    var cFileUri = cFile.uri();
     var client = newFakeClient()
-      .withInitialFs(CONFIG_SCOPE_ID, baseDir, List.of(new ClientFileDto(cppFileUri, baseDir.relativize(cppFilePath), CONFIG_SCOPE_ID, false, null, cppFilePath, null, null, true)))
+      .withInitialFs(CONFIG_SCOPE_ID, baseDir, List.of(new ClientFileDto(cFileUri, baseDir.relativize(cFilePath), CONFIG_SCOPE_ID, false, null, cFilePath, null, null, true)))
       .build();
     backend = newBackend()
       .withUnboundConfigScope(CONFIG_SCOPE_ID)
       .withStandaloneEmbeddedPluginAndEnabledLanguage(TestPlugin.CFAMILY)
       .build(client);
-    backend.getFileService().didOpenFile(new DidOpenFileParams(CONFIG_SCOPE_ID, cppFileUri));
+    backend.getAnalysisService().didSetUserAnalysisProperties(new DidChangeAnalysisPropertiesParams(CONFIG_SCOPE_ID, Map.of("sonar.cfamily.build-wrapper-content", buildWrapperContent)));
+    backend.getFileService().didOpenFile(new DidOpenFileParams(CONFIG_SCOPE_ID, cFileUri));
```

Comment:
You clean raised issues just after this line, but I don't think they will have enough time to be raised, so what you assert later might still be issues from first opening the file. I don't know how to test that better though, except waiting for issues to be raised before clearing them